services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_utils.py

```diff
@@ -0,0 +1,249 @@
+                raise e
+
+            # Fetch network name if network already exists.
+            # The environment is trashed because there seems to be an issue
+            # when stopping previous services.
+            # It is not possible to immediately remote the network after
+            # a docker-compose down involving and external overlay network
+            # has removed a container; it results as already attached
+            for network_details in await client.networks.list():
+                if network_name == network_details["Name"]:
+                    return network_details["Id"]
+
+            # finally raise an error if a network cannot be spawned
+            # pylint: disable=raise-missing-from
+            raise DynamicSidecarError(
+                f"Could not create or recover a network ID for {network_config}"
+            )
+
+
+async def create_service_and_get_id(create_service_data: Dict[str, Any]) -> str:
+    async with docker_client() as client:  # pylint: disable=not-async-context-manager
+        service_start_result = await client.services.create(**create_service_data)
+
+    if "ID" not in service_start_result:
+        raise DynamicSidecarError(
+            "Error while starting service: {}".format(str(service_start_result))
+        )
+    return service_start_result["ID"]
+
+
+async def inspect_service(service_id: str) -> Dict[str, Any]:
+    async with docker_client() as client:  # pylint: disable=not-async-context-manager
+        return await client.services.inspect(service_id)
+
+
+async def get_dynamic_sidecars_to_monitor(
+    dynamic_sidecar_settings: DynamicSidecarSettings,
+) -> Deque[ServiceLabelsStoredData]:
+    async with docker_client() as client:  # pylint: disable=not-async-context-manager
+        running_services = await client.services.list(
+            filters={
+                "label": [
+                    f"swarm_stack_name={dynamic_sidecar_settings.swarm_stack_name}"
+                ]
+            }
+        )
+
+    dynamic_sidecar_services: Deque[Tuple[str, str]] = Deque()
+
+    for service in running_services:
+        service_name: str = service["Spec"]["Name"]
+        if not service_name.startswith(DYNAMIC_SIDECAR_PREFIX):
+            continue
+
+        service_name_parts = service_name.split("_")
+        if len(service_name_parts) < 5:
+            continue
+
+        # check to see if this is a dynamic-sidecar
+        if (
+            service_name_parts[0] != DYNAMIC_SIDECAR_PREFIX
+            or service_name_parts[3] != FIXED_SERVICE_NAME_SIDECAR
+        ):
+            continue
+
+        # push found data to list
+        node_uuid = service["Spec"]["Labels"]["uuid"]
+        service_key = service["Spec"]["Labels"]["service_key"]
+        service_tag = service["Spec"]["Labels"]["service_tag"]
+        paths_mapping = PathsMappingModel(
+            **json.loads(service["Spec"]["Labels"]["paths_mapping"])
+        )
```

Comment:
there might be a function such as parse_raw or parse_obj that does this for you I think

Reply:
`parse_raw` is what I was looking for